js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -304,7 +304,43 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
             const minContextSlot = options?.minContextSlot;
             try {
                 return await this.transact(async (wallet) => {
-                    await Promise.all([
+                    function getTargetCommitment() {
```

Comment:
We should probably add this logic to Wallet Adapter's prepareTransaction method...

Reply:
😬